File: cpp/Pc.cpp
#include <iostream>

#include "Processor.cpp"
#include "Disk.cpp"
#include "Ram.cpp"

using namespace std;

class Pc
{
private:
    Processor processor;
    Disk disk;
    Ram ram;
    int totalPrice;

public:
    Pc() {}
    Pc(Processor processor, Disk disk, Ram ram)
    {
        this->processor = processor;
        this->disk = disk;
        this->ram = ram;
        this->totalPrice = processor.getPrice() + disk.getPrice() + ram.getPrice();
    }

    Processor getProcessor()
    {
        return this->processor;
    }
    void setProcessor(Processor processor)
    {
        this->processor = processor;
        this->setTotalPrice();
    }

    Disk getDisk()
    {
        return this->disk;
    }
    void setDisk(Disk disk)
    {
        this->disk = disk;
        this->setTotalPrice();
    }

    Ram getRam()
    {
        return this->ram;
    }
    void setRam(Ram ram)
    {
        this->ram = ram;
        this->setTotalPrice();
    }

    int getTotalPrice()
    {
        return this->totalPrice;
    }
    void setTotalPrice()
    {
        this->totalPrice = this->processor.getPrice() + this->disk.getPrice() + this->ram.getPrice();
    }

    void printPc()
    {
        this->processor.printProcessor();
        this->disk.printDisk();
        this->ram.printRam();
        cout << "Total Price: " << this->totalPrice << endl;
    }
    ~Pc() {}
};

File: python/Pc.py
from Processor import Processor
from Disk import Disk
from Ram import Ram

class Pc:

    def __init__(self, processor, disk, ram):
        self.processor = processor
        self.disk = disk
        self.ram = ram
        self.totalPrice = processor.getPrice() + disk.getPrice() + ram.getPrice()

    def getProcessor(self):
        return self.processor
    
    def setProcessor(self, processor):
        self.processor = processor

    def getDisk(self):
        return self.disk

    def setDisk(self, disk):
        self.disk = disk

    def getRam(self):
        return self.ram

    def setRam(self, ram):
        self.ram = ram

    def getTotalPrice(self):
        return self.totalPrice

    def setTotalPrice(self):
        self.totalPrice = self.processor.getPrice() + self.disk.getPrice() + self.ram.getPrice()

    def printPc(self):
        self.processor.printProcessor()
        self.disk.printDisk()
        self.ram.printRam()
        print("Total Price: {}".format(self.totalPrice))